deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/FileUploadCompletionNotification.java

```diff
@@ -0,0 +1,195 @@
+     * @throws IllegalArgumentException if one of the parameters is null, empty, or not valid.
+     */
+    public FileUploadCompletionNotification(String correlationId, Boolean isSuccess)
+            throws IllegalArgumentException
+    {
+        setCorrelationId(correlationId);
+        updateStatus(isSuccess, statusCode, statusDescription);
+    }
+
+    /**
+     * Create an instance of the FileUploadCompletionNotification for a single file upload operation using Azure Storage.
+     *
+     * @param correlationId the correlationId that correlates this FileUploadCompletionNotification to the earlier request to get the SAS URI
+     *                      for this upload from IoT Hub. This field is mandatory. Must equal {@link FileUploadSasUriResponse#getCorrelationId()}.
+     * @param isSuccess whether the file was uploaded successfully. This field is mandatory.
+     * @param statusCode is the status for the upload of the file to storage.
+     * @param statusDescription is the description of the status code.
+     * @throws IllegalArgumentException if one of the parameters is null, empty, or not valid.
+     */
+    public FileUploadCompletionNotification(String correlationId, Boolean isSuccess, Integer statusCode, String statusDescription)
+            throws IllegalArgumentException
+    {
+        setCorrelationId(correlationId);
+        updateStatus(isSuccess, statusCode, statusDescription);
+    }
+
+    /**
+     * Construct this notification with json
+     * @param json the json to parse.
+     */
+    public FileUploadCompletionNotification(String json)
+    {
+        Gson gson = new GsonBuilder().disableHtmlEscaping().serializeNulls().create();
+        FileUploadCompletionNotification fileUploadCompletionNotification;
+
+        try
+        {
+            fileUploadCompletionNotification = gson.fromJson(json, FileUploadCompletionNotification.class);
+        }
+        catch (Exception malformed)
+        {
+            throw new IllegalArgumentException("Malformed json:" + malformed);
+        }
+
+        this.correlationId = fileUploadCompletionNotification.getCorrelationId();
+        this.isSuccess = fileUploadCompletionNotification.getSuccess();
+        this.statusCode = fileUploadCompletionNotification.getStatusCode();
+        this.statusDescription = fileUploadCompletionNotification.getStatusDescription();
+    }
+
+    /**
+     * Update the status information in the collection, and return the new json.
+     *
+     * @param isSuccess is a Boolean representing whether the file was uploaded successfully.
+     * @param statusCode is the status for the upload of the file to storage.
+     * @param statusDescription is the description of the status code.
+     * @throws IllegalArgumentException if one of the parameters is null, empty, or not valid.
+     */
+    private void updateStatus(Boolean isSuccess, Integer statusCode, String statusDescription) throws IllegalArgumentException
```

Comment:
if the first constructor is used to create this object, the statusDescription is null. So wouldn't this method throw since statusDescription is null?

Reply:
See above, I'll delete this javadoc